src/c++/perf_analyzer/test_dataloader.cc

```diff
@@ -0,0 +1,576 @@
+//    doctest::description(
+//        "If the size of the provided Input is not in line with the Tensor's "
+//        "shape, then an error should be thrown"))
+//{
+//  std::string json_str{R"({"data": [{ "INPUT1": [1,2] }]})"};
+//
+//
+//  MockDataLoader dataloader;
+//  std::shared_ptr<ModelTensorMap> inputs = std::make_shared<ModelTensorMap>();
+//  ModelTensor input1 = TestDataLoader::GetTensor("INPUT1");
+//  input1.shape_ = {3};
+//  inputs->insert(std::make_pair(input1.name_, input1));
+//
+//  std::shared_ptr<ModelTensorMap> outputs =
+//  std::make_shared<ModelTensorMap>();
+//
+//  cb::Error status = dataloader.ReadDataFromJSON(inputs, outputs, json_str);
+//  CHECK_FALSE(status.IsOk());
+//  CHECK_EQ(status.Message(), "TKG");
+//}
+
+
+TEST_CASE(
+    "dataloader: ReadDataFromJSON: Mismatch Input and Output" *
+    doctest::description(
+        "If the size of the provided Input and validation Output data are "
+        "different, then an error should be thrown"))
+{
+  std::string json_str{R"({
+   "data": [
+     { "INPUT1": [1] },
+     { "INPUT1": [2] },
+     { "INPUT1": [3] }
+   ],
+   "validation_data": [
+     { "OUTPUT1": [7] }
+   ]})"};
+
+
+  MockDataLoader dataloader;
+  std::shared_ptr<ModelTensorMap> inputs = std::make_shared<ModelTensorMap>();
+  std::shared_ptr<ModelTensorMap> outputs = std::make_shared<ModelTensorMap>();
+
+  cb::Error status = dataloader.ReadDataFromJSON(inputs, outputs, json_str);
+  CHECK_FALSE(status.IsOk());
+  CHECK_EQ(
+      status.Message(),
+      "The 'validation_data' field doesn't align with 'data' field in the json "
+      "file");
+}
+
+TEST_CASE("dataloader: ReadDataFromJSON: Valid Data")
+{
+  std::string json_str{R"({
+   "data": [
+     { "INPUT1": [1] },
+     { "INPUT1": [2] },
+     { "INPUT1": [3] }
+   ],
+   "validation_data": [
+    { "OUTPUT1": [4] },
+    { "OUTPUT1": [5] },
+    { "OUTPUT1": [6] }
+   ]})"};
+
+  MockDataLoader dataloader;
+  std::shared_ptr<ModelTensorMap> inputs = std::make_shared<ModelTensorMap>();
+  std::shared_ptr<ModelTensorMap> outputs = std::make_shared<ModelTensorMap>();
+
+  ModelTensor input1 = TestDataLoader::GetTensor("INPUT1");
+  ModelTensor output1 = TestDataLoader::GetTensor("OUTPUT1");
+
+  inputs->insert(std::make_pair(input1.name_, input1));
+  outputs->insert(std::make_pair(output1.name_, output1));
+
+  cb::Error status = dataloader.ReadDataFromJSON(inputs, outputs, json_str);
+  REQUIRE(status.IsOk());
+  CHECK_EQ(dataloader.GetDataStreamsCount(), 1);
+  CHECK_EQ(dataloader.GetTotalSteps(0), 3);
+
+  // Confirm the correct data is in the dataloader
+  //
+  const uint8_t* data_ptr{nullptr};
+  size_t batch_size;
+  std::vector<int64_t> shape;
+
+  dataloader.GetInputShape(input1, 0, 1, &shape);
+  CHECK_EQ(shape.size(), 1);
+  CHECK_EQ(shape[0], 1);
+
+  dataloader.GetInputData(input1, 0, 1, &data_ptr, &batch_size);
+  auto data = *reinterpret_cast<const int32_t*>(data_ptr);
+  CHECK_EQ(data, 2);
+  CHECK_EQ(batch_size, 4);
```

Comment:
Why is this called batch size?
Is this one of your refactor items?

Reply:
it is batch1_size now